Source/WebKit/WebProcess/WebStorage/StorageNamespaceImpl.cpp

```diff
@@ -42,19 +42,20 @@
 namespace WebKit {
 using namespace WebCore;
 
-static HashMap<StorageNamespaceImpl::Identifier, StorageNamespaceImpl*>& sessionStorageNamespaces()
+static HashMap<std::pair<StorageNamespaceImpl::Identifier, WebCore::SecurityOriginData>, StorageNamespaceImpl*>& sessionStorageNamespaces()
 {
-    static NeverDestroyed<HashMap<StorageNamespaceImpl::Identifier, StorageNamespaceImpl*>> map;
+    static NeverDestroyed<HashMap<std::pair<StorageNamespaceImpl::Identifier, WebCore::SecurityOriginData>, StorageNamespaceImpl*>> map;
     return map;
 }
 
-Ref<StorageNamespaceImpl> StorageNamespaceImpl::createSessionStorageNamespace(Identifier identifier, PageIdentifier pageID, unsigned quotaInBytes)
+RefPtr<StorageNamespaceImpl> StorageNamespaceImpl::maybeCreateSessionStorageNamespace(Identifier identifier, PageIdentifier pageID, WebCore::SecurityOrigin& topLevelOrigin, unsigned quotaInBytes, bool optionalCreate)
```

Comment:
Nothing keeps SessionStorageNamespace alive for the issue mentioned above; so this function will always create new SessionStorageNamespace I think, if optionalCreate is true.